src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -157,7 +157,15 @@ public ConcurrentDictionary(int concurrencyLevel, int capacity, IEqualityCompare
 
         internal ConcurrentDictionary(int concurrencyLevel, int capacity, bool growLockArray, IEqualityComparer<TKey>? comparer)
         {
-            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(concurrencyLevel);
+            if (concurrencyLevel == -1)
+            {
+                concurrencyLevel = DefaultConcurrencyLevel;
+            }
+            else
+            {
+                ArgumentOutOfRangeException.ThrowIfNegativeOrZero(concurrencyLevel);
```

Comment:
The exception message will be slightly misleading. It doesn't seem worth the cost of carrying a new string forever and always though. Was that your thinking too?

Reply:
Yes.  But since you called it out, too, I've changed it.  Turns out we actually had an unused resource intended for this case, so I just adapted it.